### PR TITLE
go/libraries/doltcore/remotestorage: Move retrying and metrics instrumentation for doltremoteapi interactions to grpc client interceptors.

### PR DESCRIPTION
**Pull request by reltuk**

### CODE REVIEW

File: go/cmd/dolt/commands/login.go [hunk 1]
```diff
@@ -198,7 +200,15 @@ func openBrowserForCredsAdd(dEnv *env.DoltEnv, dc creds.DoltCreds) {
 func getCredentialsClient(dEnv *env.DoltEnv, dc creds.DoltCreds) (remotesapi.CredentialsServiceClient, errhand.VerboseError) {
 	host := dEnv.Config.GetStringOrDefault(env.RemotesApiHostKey, env.DefaultRemotesApiHost)
 	port := dEnv.Config.GetStringOrDefault(env.RemotesApiHostPortKey, env.DefaultRemotesApiPort)
-	conn, err := dEnv.GrpcConnWithCreds(fmt.Sprintf("%s:%s", *host, *port), false, dc)
+	endpoint, opts, err := dEnv.GetGRPCDialParams(grpcendpoint.Config{
+		Endpoint: fmt.Sprintf("%s:%s", *host, *port),
+		Insecure: false,
```
Comment (bheni) [inline go/cmd/dolt/commands/login.go:205]: Above you use the zero value, here you explicitly use false.  I would choose one or the other.

File: go/libraries/doltcore/remotestorage/retrying_interceptor.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"google.golang.org/grpc"
+)
+
+const (
+	csClientRetries = 5
+)
+
+var csRetryParams = backoff.NewExponentialBackOff()
+
+func RetryingUnaryClientInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```
Comment (bheni) [inline go/libraries/doltcore/remotestorage/retrying_interceptor.go:30]: sweet

File: go/libraries/doltcore/remotestorage/events_interceptor.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	eventsapi "github.com/liquidata-inc/dolt/go/gen/proto/dolt/services/eventsapi/v1alpha1"
+	"github.com/liquidata-inc/dolt/go/libraries/events"
+)
+
+func EventsUnaryClientInterceptor(collector *events.Collector) grpc.UnaryClientInterceptor {
```
Comment (bheni) [inline go/libraries/doltcore/remotestorage/events_interceptor.go:26]: @coffeegoddd FYI.  Take a look at remotestorage/chunk_store.go

Comment (coffeegoddd) [reply go/libraries/doltcore/remotestorage/events_interceptor.go:26]: much improvement

**Pull request merged**